app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -21,55 +32,41 @@ class SearchActivity : AppCompatActivity() {
+    private lateinit var searchPlaceholderErrorIcon: ImageView
+    private lateinit var searchPlaceholderErrorText: TextView
+    private lateinit var searchPlaceholderRefreshButton: Button
+
+    var searchText: String? = null
 
     private lateinit var trackListView: RecyclerView
 
+    private val iTunesBaseUrl = "https://itunes.apple.com"
+
+    private val retrofit = Retrofit.Builder()
+        .baseUrl(iTunesBaseUrl)
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
+
+    // Добавление треков в список
+    val tracks: MutableList<Track> = mutableListOf()
+
+    private val iTunesService = retrofit.create(ITunesAPI::class.java)
+    private val adapter = TrackAdapter(tracks)
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_search)
 
-        trackListView = findViewById(R.id.trackList)
+        searchPlaceholder = findViewById(R.id.searchPlaceholder)
+        searchPlaceholderErrorIcon = findViewById(R.id.searchPlaceholderErrorIcon)
+        searchPlaceholderErrorText = findViewById(R.id.searchPlaceholderErrorText)
+        searchPlaceholderRefreshButton = findViewById(R.id.searchPlaceholderRefreshButton)
+
+        trackListView = findViewById(R.id.trackListView)
         trackListView.layoutManager = LinearLayoutManager(this)
 
-        // Добавление треков в список
-        val trackList: MutableList<Track> = mutableListOf()
-        trackList.add(Track(
-            "Smells Like Teen Spirit",
-            "Nirvana",
-            "5:01",
-            "https://is5-ssl.mzstatic.com/image/thumb/Music115/v4/7b/58/c2/7b58c21a-2b51-2bb2-e59a-9bb9b96ad8c3/00602567924166.rgb.jpg/100x100bb.jpg"
-        ))
-
-        trackList.add(Track(
-            "Billie Jean",
-            "Michael Jackson",
-            "4:35",
-            "https://is5-ssl.mzstatic.com/image/thumb/Music125/v4/3d/9d/38/3d9d3811-71f0-3a0e-1ada-3004e56ff852/827969428726.jpg/100x100bb.jpg"
-        ))
-
-        trackList.add(Track(
-            "Stayin' Alive",
-            "Bee Gees",
-            "4:10",
-            "https://is4-ssl.mzstatic.com/image/thumb/Music115/v4/1f/80/1f/1f801fc1-8c0f-ea3e-d3e5-387c6619619e/16UMGIM86640.rgb.jpg/100x100bb.jpg"
-        ))
-
-        trackList.add(Track(
-            "Whole Lotta Love",
-            "Led Zeppelin",
-            "5:33",
-            "https://is2-ssl.mzstatic.com/image/thumb/Music62/v4/7e/17/e3/7e17e33f-2efa-2a36-e916-7f808576cf6b/mzm.fyigqcbs.jpg/100x100bb.jpg"
-        ))
-
-        trackList.add(Track(
-            "Sweet Child O'Mine",
-            "Guns N' Roses",
-            "5:03",
-            "https://is5-ssl.mzstatic.com/image/thumb/Music125/v4/a0/4d/c4/a04dc484-03cc-02aa-fa82-5334fcb4bc16/18UMGIM24878.rgb.jpg/100x100bb.jpg"
-        ))
-
-        val trackAdapter = TrackAdapter(trackList)
+        val trackAdapter = TrackAdapter(tracks)
         trackListView.adapter = trackAdapter
```

Comment:
Для списка устанавливается адаптер `trackAdapter` вместо `adapter`

```suggestion
        trackListView.adapter = adapter
```

Reply:
Все поправил, спасибо!